backend/controller/deployment_logs.go

```diff
@@ -0,0 +1,85 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"github.com/TBD54566975/ftl/backend/common/log"
+	"github.com/TBD54566975/ftl/backend/common/model"
+	"github.com/TBD54566975/ftl/backend/controller/internal/dal"
+	"github.com/alecthomas/errors"
+	"github.com/alecthomas/types"
+	"time"
+)
+
+type logEntry struct {
+	request types.Option[model.IngressRequestKey]
+	log.Entry
+}
+
+var _ log.Sink = (*deploymentLogsSink)(nil)
+
+func newDeploymentLogsSink(ctx context.Context, key model.DeploymentKey, dal *dal.DAL) *deploymentLogsSink {
+	sink := &deploymentLogsSink{
+		deploymentKey: key,
+		logQueue:      make(chan logEntry, 100),
+		dal:           dal,
+	}
+
+	// Process logs in background
+	go sink.processLogs(ctx)
+
+	return sink
+}
+
+type deploymentLogsSink struct {
+	deploymentKey model.DeploymentKey
+	logQueue      chan logEntry
+	dal           *dal.DAL
+}
+
+// Log implements Sink
+func (d *deploymentLogsSink) Log(entry log.Entry) error {
+	var request types.Option[model.IngressRequestKey]
+	if reqStr, ok := entry.Attributes["request"]; ok {
+		req, err := model.ParseIngressRequestKey(reqStr)
+		if err == nil {
+			request = types.Some(req)
+		}
+	}
```

Comment:
Not sure if this is needed. Will we have `request` attributes here?

Reply:
I concluded that we'll have to have it in the logs, because we can't get it from anywhere else as the logger doesn't accept a Context.

TL;DR I couldn't think of a better idea, and I think this makes sense as we're doing it for other keys.